geo/src/main/java/org/apache/carbondata/geo/scan/filter/executor/PolygonFilterExecutorImpl.java

```diff
@@ -73,8 +77,24 @@ private int getNearestRangeIndex(List<Long[]> ranges, long searchForNumber) {
    * @return True or False  True if current block or blocket needs to be scanned. Otherwise False.
    */
   private boolean isScanRequired(byte[] maxValue, byte[] minValue) {
-    PolygonExpression polygon = (PolygonExpression) exp;
-    List<Long[]> ranges = polygon.getRanges();
+    List<Long[]> ranges = new ArrayList<>();
+    if (exp instanceof PolygonExpression) {
```

Comment:
These changes wouldn't be required if all new expressions had extended `PolygonExpression`

Reply:
Removed the changes.